app/Main.hs

```diff
@@ -1,6 +1,17 @@
 module Main where
 
-import Lib
+import Control.Monad (join)
+
+import Options.Applicative
+
+import Cache (cacheBoards)
+import Search (searchBoards)
+
+args :: Parser (IO ())
+args =
+    subparser $
+        command "search" (info (searchBoards <$> (optional $ argument str (metavar "QUERY"))) idm) <>
```

Comment:
Usually I don't mix `$` and parentheses -- in this case I'd use all parentheses.

Or, even better, define some `where`s:

```haskell
args = subparser $ searchCommand <> cacheCommand 
    where
        searchCommand =  command "search" (info (searchBoards <$> (optional $ argument str (metavar "QUERY"))) idm)
        cacheCommand =  command "cache" (info (pure cacheBoards) (progDesc "foo"))
```